llvm/include/llvm/Analysis/ConstraintSystem.h

```diff
@@ -51,7 +51,7 @@ class ConstraintSystem {
 
   /// A map of variables (IR values) to their corresponding index in the
   /// constraint system.
-  DenseMap<Value *, unsigned> Value2Index;
+  SmallDenseMap<Value *, unsigned, 16> Value2Index;
```

Comment:
I'd add a using/typedef for this one, especially as it needs to be synchronized with ConstraintElimination.